repoconfig.sh

```diff
@@ -1,6 +1,15 @@
 #! /bin/sh
 # shellcheck disable=SC2034
 NODEJS_VERSION=v16
-GOLANG_VERSION=1.19
+GOLANG_VERSION=1.20
 GOLANG_DIR=golang/cosmos
 GOLANG_DAEMON=$GOLANG_DIR/build/agd
+
+# Args are major, minor and patch version numbers
+function golang_version_check() {
+  # Want 1.20.2+, 1.21+, or 2+
+  [ $1 -eq 1 ] && {
+    [ $2 -lt 20 ] && echo "need go version 1.20 or higher" 1>&2 && return 1
+    [ $2 -eq 20 ] && [ $3 -lt 2 ] && echo "need go version 1.20.2 or higher" 1>&2 && return 1
+  } || true
```

Comment:
We can make the shell more like a Go switch:

```suggestion
  if [ $1 -ne 1 ]; then :
  elif [ $2 -lt 20 ]; then :
  elif [ $2 -eq 20 ] && [ $3 -lt 2 ]; then : 
  else
    return 0
  fi
  echo "need go version 1.20.2 or higher" 1>&2
  return 1
```

Reply:
Imagine we were still allowing 1.19.7+, or adding a constraint of 1.21.1+ - the logic here would get pretty janky.

I've flattened it to make it as simple and clear as I can think of without getting too repetitious.